### ISSUE TITLE
screen reader scan mode: checkbox focused multi-times

### ISSUE THREAD

**Issue by YoHo-Wang** [microsoft/fluentui#14192]
### Environment Information

- **Package version(s)**: "office-ui-fabric-react": "7.122.0"
- **Browser and OS versions**: Microsoft Edge
Microsoft Edge is up to date.
Version 85.0.564.13 (Official build) dev (64-bit)

### Please provide a reproduction of the issue in a codepen:
checkbox demo in this site:
https://developer.microsoft.com/en-us/fluentui#/controls/web/checkbox

#### Actual behavior:
1. press "ctrl"+"win"+"enter" open screen reader. 
2. press "capslock"+"space" open scan mode.
3. use arrow key go through checkbox component. it will go 3 times for each checkbox item.

#### Expected behavior:
each checkbox item go through 1 time when in scan mode.

**Comment by paulgildea** [comment]
@smhigley FYI

**Comment by smhigley** [comment]
There are two separate issues going on here, one of which is on Fluent, and one on Edge/Narrator (probably Narrator).

The Fluent issue is that the `<label>` elements here have had all their text hidden, and label element does require an accessible name (see "name required" here: https://w3c.github.io/aria/#label). This could be fixed just by removing `aria-hidden` on the text node within the label. (and the aria-label on the checkbox input isn't necessary and is slightly worse for 3rd-party language translation when present, but the label text should be available regardless). This affects more than just the Narrator bug filed -- multiple screen readers don't read the accessible name of the checkbox input if using arrow keys to move through elements, under the expectation that the text of the accessible name is likely adjacent. If the text is not adjacent and exposed in the a11y tree, that breaks this experience.

The Narrator/Edge issue is that Narrator is adding empty accessible nodes that result in scan mode stops for the empty ::before CSS pseudo-element, and the empty div that renders the checkbox icon. This is a pretty straightforward browser or screen reader bug (I think it's Narrator based on looking at the UIA tree, but not 100% sure)

**Comment by DrewCanitz** [comment]
I also got an accessibility bug for my product matching the above bug description.
